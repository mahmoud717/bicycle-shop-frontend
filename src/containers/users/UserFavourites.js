import { React, useEffect, useState } from 'react';

import { Link, useParams, useHistory } from 'react-router-dom';
import axios from 'axios';

const UserFavorites = () => {
  const { id } = useParams();
  const history = useHistory();
  const [favorites, setFavorites] = useState([]);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    axios.get(`https://bicycle-shop-backend.herokuapp.com/api/v1/users/${id}/favourites`)
      .then(response => {
        setFavorites(response.data);
        setLoading(true);
      })
      .catch(() => {
        history.pushState('/404');
      });
  }, []);
  if (loading) {
    <h1 className="text-center mt-5">loading</h1>;
  }
  if (!loading && !favorites.length) {
    return (
      <h1 className="text-center mt-5">There are no favourites</h1>
    );
  }
  return (
    <div className="container-fluid order-card-container d-flex flex-column justify-content-center align-items-center">
      {favorites.map(bike => (
        <Link to={`/bicycles/${bike.id}`} key={bike.id}>
          <div className="order-card d-flex">
            <div className="order-card-image">
              <img src={bike.image_url} alt="" className="w-100" />
            </div>
            <div className="order-card-details">
              <div className="order-card-name">{bike.name}</div>
              <div className="order-card-data">{bike.created_at}</div>
            </div>
          </div>
        </Link>
      ))}
    </div>
  );
};

export default UserFavorites;
