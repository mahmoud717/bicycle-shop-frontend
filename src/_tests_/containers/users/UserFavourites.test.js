import React from 'react';

import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import UserFavourites from '../../../containers/users/UserFavourites';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <UserFavourites />
  </Provider>,
);

describe('UserFavourites Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the UserFavourites container', () => {
    const page = component.find('.user-favourites');
    expect(page).toMatchSnapshot();
  });
});
