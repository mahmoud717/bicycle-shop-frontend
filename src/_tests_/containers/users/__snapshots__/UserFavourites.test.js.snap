// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFavourites Component renders the UserFavourites container 1`] = `ShallowWrapper {}`;
